app/src/forms/form/controller.js

```diff
@@ -387,4 +387,52 @@ module.exports = {
       next(error);
     }
   },
+  listExternalAPIs: async (req, res, next) => {
+    try {
+      const response = await service.listExternalAPIs(req.params.formId);
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  },
+  listExternalAPIAlgorithms: async (req, res, next) => {
+    try {
+      const response = await service.listExternalAPIAlgorithms();
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  },
+  listExternalAPIStatusCodes: async (req, res, next) => {
+    try {
+      const response = await service.listExternalAPIStatusCodes();
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  },
+  createExternalAPI: async (req, res, next) => {
+    try {
+      const response = await service.createExternalAPI(req.params.formId, req.body, req.currentUser);
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  },
+  updateExternalAPI: async (req, res, next) => {
+    try {
+      const response = await service.updateExternalAPI(req.params.formId, req.params.externalAPIId, req.body, req.currentUser);
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  },
+  deleteExternalAPI: async (req, res, next) => {
+    try {
+      const response = await service.deleteExternalAPI(req.params.formId, req.params.externalAPIId);
+      res.status(200).json(response);
```

Comment:
For the delete I don't think `service.deleteExternalAPI` returns anything, so there is no `response` to send? I think for deletes we usually return a 204 response. (And again I think this is something that probably needs to be fixed in other places in the controllers!)
```suggestion
      await service.deleteExternalAPI(req.params.formId, req.params.externalAPIId);
      res.status(204).send();
```

Reply:
Oh, and once the return values are figured out we do have OpenAPI docs in `app/src/docs/v1.api-spec.yaml`. It's available as http://localhost:8080/app/api/v1/docs